### ISSUE TITLE
Creating a forced download link

### ISSUE THREAD

**Issue by ZempTime** [refile/refile#118]
This might be a super basic question but it's late and I'm stumped. I'm using Refile to serve up audio files uploaded by musicians:

``` ruby
class Song < ActiveRecord::Base
  attachment :audio_file
end
```

I'm also storing the _filename _size _content_type rows in my songs table

I want to create a link so that when someone clicks it, it automatically downloads. Currently, I can get it to open streaming via attachment_url, but not the equivalent of "save target as". I'm not sure how to serve this out from the controller in a send_file, either.

``` ruby
#View
  <%= link_to "download", attachment_url(song, :audio_file, filename: song.audio_file_filename)%>  #opens up an mp3 player
  <%= link_to "controller download ", user_album_song_download_path(current_user, @album, song) %> 

#controller - not sure what to do here, either
  def download
    send_file Refile.attachment_url(@song, :audio_file), type: @song.audio_file_content_type, disposition: :attachment
  end
```

Where am I going wrong, and is this in the documentation anywhere?

**Comment by excid3** [comment]
It seems wise to make the disposition `:inline` by default but make it an option we can pass in to the Sinatra app.

https://github.com/elabs/refile/blob/master/lib/refile/app.rb#L99

**Comment by trenchwarfare** [comment]
:+1:  your branch is working! thanks @excid3

**Comment by paulstraw** [comment]
Hey all, just checking in on this issue. I just saw #119 and @jnicklas' [comment](https://github.com/refile/refile/pull/119#issuecomment-70390945) on it. Is anyone already working on this? I'd be happy to tackle it if @jnicklas is happy with that solution, or has another in mind.

**Comment by matiasmasca** [comment]
When I try to do the same I received the message "
ActionController::MissingFile at /attachments/download

Cannot read file http://0.0.0.0:3000/attachments/5766873402ffbc33fba841ec36eed1bd399b4641/store/d7f124d05a7abf06ae40cf43d317f6dff8c6417dd80d0896f1e98f055990/200x200tmdg.jpg"

but If I open that URL in a new tab I can see the file

**Comment by matiasmasca** [comment]
Finally I resolved it... in this way... 

file = Refile.store.get(@attachment.profile_image_id).download

send_file file, filename: @attachment.profile_image_filename, type: @attachment.profile_image_content_type, disposition: :attachment

**Comment by mtrolle** [comment]
@matiasmasca thanks - your solution worked well for me! :+1:

**Comment by chevinbrown** [comment]
@matiasmasca, Trying to understand how to provide direct download links with latest refile version. Is this still applicable? Can you maybe expand your solution to help us less developed learners?

**Comment by senny** [comment]
Is the recommended workaround still to send the file in a custom controller action?

**Comment by sobrinho** [comment]
Closed by #561.
